### PR TITLE
Parse backend options in backend

### PR DESCRIPTION
**Pull request by qwerty287**
Currently, backend options are parsed in the yaml parser.
This has some issues:
- backend specific code should be in the backend folders
- it is not possible to add backend options for backends added via addons

### CODE REVIEW

**Review by zc-devs — COMMENTED** [review]
I like the idea in general 👍 

> added via plugins

Did you mean `addons`?

**Review by zc-devs — APPROVED** [review]

File: pipeline/backend/kubernetes/backend_options.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package kubernetes
+
+import (
+	"github.com/mitchellh/mapstructure"
```
Comment (6543) [inline pipeline/backend/kubernetes/backend_options.go:4]: without looking into the lib - why was it used ?

**Review by anbraten — APPROVED** [review]
Apart from the discussions regarding the data format passed to the backend, code lgtm

**Comment by 6543** [comment]
Yes that concerns me too ... but i dont have time to look into it so, lgtm

**Comment by anbraten** [comment]
Just to clarify. lgtm. Just wanted to ping as 6543 requested changes.

**Pull request merged**